source/PlayState.hx

```diff
@@ -89,11 +121,14 @@ class PlayState extends FlxState {
 	override public function update(elapsed:Float) {
 		super.update(elapsed);
 
+		FlxG.overlap(player, obstacleGeneratorBot.obstacles, DeadlyObstacle.overlapsWithPlayer);
+		FlxG.overlap(player, obstacleGeneratorMid.obstacles, DeadlyObstacle.overlapsWithPlayer);
+		FlxG.overlap(player, obstacleGeneratorTop.obstacles, DeadlyObstacle.overlapsWithPlayer);
```

Comment:
If we're using `DeadlyObstacle`'s overlap resolution method, let's just make all of these `DeadlyObstacles` (or some sub-class of it). Having visual variation between these vertical ones and the horizontal ones is nice, so I think it'd be good to make that possible with whatever solution we end up on.